src/explore-education-statistics-common/src/modules/charts/util/getMinorAxisSize.ts

```diff
@@ -0,0 +1,38 @@
+import { DataSetCategory } from '@common/modules/charts/types/dataSet';
+import parseNumber from '@common/utils/number/parseNumber';
+
+interface Props {
+  dataSetCategories: DataSetCategory[];
+  minorAxisSize: number | undefined;
+  minorAxisDecimals: number | undefined;
+  minorAxisUnit: string;
+}
+
+/**
+ * Get the highest value used on the axis and determine the axis width based on its length.
+ * 12 is a fairly arbitrary width here, more than needed for just the numbers to take into account commas etc.
+ */
+const getMinorAxisSize = ({
+  dataSetCategories,
+  minorAxisSize,
+  minorAxisDecimals = 0,
+  minorAxisUnit,
+}: Props) => {
+  if (minorAxisSize) {
+    return parseNumber(minorAxisSize);
+  }
+
+  const highestValueLength = dataSetCategories
+    .reduce((acc, dataSetCategory) => {
+      const value = Object.values(dataSetCategory.dataSets).reduce(
+        (acc2 = acc, dataSet) => (dataSet.value > acc2 ? dataSet.value : acc2),
+        0,
+      );
+      return value > acc ? value : acc;
+    }, 0)
+    .toString().length;
```

Comment:
I don't know if this is a real-world scenario or not, but if we had 2 number values in our `dataSetCategories` like:

`9`

and

`3.5`

I guess this reduction could potentially identify `9` as the highest value and use that to calculate the length against (so 12 in this case), but `3.5` is actually the *longest* number string and so should ideally yield a length of 36.

Again though, I'm not sure if this would ever happen?

Reply:
Adding the minorAxisDecimals handles this.